packages/core/src/Machine.ts

```diff
@@ -1,44 +1,21 @@
 import {
   MachineImplementations,
-  DefaultContext,
   MachineConfig,
-  StateSchema,
   EventObject,
   AnyEventObject,
   Typestate
 } from './types';
 import { MachineNode } from './MachineNode';
 import { Model, ModelContextFrom, ModelEventsFrom } from './model';
 
-export function Machine<
-  TContext = any,
-  TEvent extends EventObject = AnyEventObject
->(
-  definition: MachineConfig<TContext, TEvent, any>,
-  implementations?: Partial<MachineImplementations<TContext, TEvent>>
-): MachineNode<TContext, TEvent>;
-export function Machine<
-  TContext = DefaultContext,
-  TEvent extends EventObject = AnyEventObject,
-  TStateSchema extends StateSchema = any
->(
-  definition: MachineConfig<TContext, TEvent, TStateSchema>,
-  implementations?: Partial<MachineImplementations<TContext, TEvent>>
-): MachineNode<TContext, TEvent, TStateSchema>;
-export function Machine<
-  TContext = DefaultContext,
+export function createMachine<
+  TContext extends never,
```

Comment:
q: what's the purpose of this overload?

Reply:
I should add a comment... `any` extends `never`, so if the developer specifies `createMachine<any, any>`, it will correctly choose the context-event overload instead of the model overload.